README.md
# apue
test code for apue


02.chapter/history.txt
    1  ll
    2  g apue
    3  cd code
    4  ll
    5  make
    6  cd apue
    7  make
    8  make clean
    9  make
   10  ll
   11  cd ..
   12  mkdir tools
   13  cd tools
   14  mv ~/Desktop/bashrc .
   15  mv ~/Desktop/vimrc .
   16  mv ~/Desktop/bashmarks-master.tar.gz .
   17  ll
   18  tar xzvf bashmarks-master.tar.gz .
   19  ll
   20  tar xzvf bashmarks-master.tar.gz 
   21  ll
   22  cd bashmarks-master
   23  ll
   24  make 
   25  make install
   26  ll
   27  cd 
   28  cd -
   29  cd ..
   30  ls
   31  diff vimrc /etc/vimrc
   32  diff bashrc ~/.bashrc
   33  exit
   34  ll
   35  cd ..
   36  vi .bashrc
   37  . .bashrc
   38  l
   39  cd code/apue
   40  s apue
   41  cd ../akae
   42  s akae
   43  cd ~
   44  g akae
   45  g apue
   46  g ..
   47  ll
   48  cd ..
   49  l
   50  vi /etc/vimrc
   51  sudo vi /etc/vimrc
   52  ll
   53  cd tools
   54  ll
   55  vi ~/.bashrc
   56  . ~/.bash
   57  . ~/.bashrc
   58  cd ..
   59  vi /etc/vimrc
   60  sudo vi /etc/vimrc
   61  ll
   62  cd code
   63  cd apue
   64  cd 02.chapter/
   65  ll
   66  make clean
   67  vi seekhole.c
   68  ll
   69  vi read_inter.c 
   70  cp read_inter.c write_seek.c
   71  vi seekable.c
   72  vi seekhole.c
   73  make seekhole
   74  ./seekhole
   75  ll
   76  vi file.hole
   77  vi file.nohole 
   78  vi seekhole.c
   79  make clean
   80  l
   81  ls
   82  ll
   83  vi write_seek.c 
   84  vi Makefile 
   85  make
   86  ll
   87  ./write_seek file.append
   88  ll
   89  vi write_seek.c
   90  write_seek 
   91  ./write_seek
   92  ./write_seek file.app
   93  strace ./write_seek file.app
   94  vi write_seek.c
   95  vi read_inter.c
   96  make
   97  ./write_seek file.app
   98  vi write_seek.c
   99  make
  100  ./write_seek file.app
  101  ll
  102  ls -lhrt
  103  rm file.app
  104  ls -lhrt
  105  vi write_seek.c
  106  make 
  107  ./write_seek  file.app
  108  ll
  109  ls -lhrt
  110  vi file.app
  111  sudo vi file.app
  112  rm file.app
  113  vi write_seek.c
  114  make
  115  ./write_seek file.app
  116  ls -lhrt
  117  vi file.app
  118  vi write_seek.c
  119  rm file.app
  120  ll
  121  make
  122  vi write_seek.c
  123  make
  124  ./write_seek file.app
  125  ls -lhrt
  126  vi file.app
  127  vi write_seek.c
  128  make
  129  ./write_seek file.app
  130  vi file.app
  131  vi write_seek.c
  132  make
  133  rm file.app
  134  ./write_seek file.app
  135  vi file.app
  136  ls -lhrt
  137  vi write_seek.c
  138  make
  139  vi write_seek.c
  140  make
  141  rm file.app
  142  ./write_seek file.app
  143  vi file.app
  144  ls -lhrt
  145  ls -l
  146  vi creat.c
  147  ls -lhrts
  148  ls -lhts
  149  ls -lrts
  150  vi write_seek.c
  151  make
  152  vi write_seek.c
  153  make
  154  vi write_seek.c
  155  make
  156  ls 
  157  rm file.app
  158  ./write_seek file.app
  159  vi file.app
  160  vi write_seek.c
  161  make
  162  ./write_seek file.app
  163  cd 
  164  cd -
  165  cd --
  166  g apue
  167  ls
  168  grep RLIMIT_FSIZE . -R -color
  169  grep RLIMIT_FSIZE . -R
  170  vi 01.chapter/limit.c
  171  cd 01.chapter/
  172  make
  173  ./limit 
  174  limit
  175  vi ~/.bashrc
  176  vi ~/.local/bin/bashmarks.sh 
  177  . ~/.bashrc
  178  g akae
  179  exit
  180  g akae
  181  g apue
  182  cd 02.chapter/
  183  ls
  184  cp write_seek.c write_limit.c
  185  vi write_limit.c 
  186  vi Makefile 
  187  make
  188  vi write_limit.c 
  189  make
  190  ./write_limit file.app
  191  ll
  192  ls -lhrt
  193  ulimit -f 512K -S
  194  ulimit -f 512000 -S
  195  ulimit -a
  196  ls -l
  197  rm file.app
  198  ./write_limit file.app
  199  ll
  200  ulimit -f 512000 -H
  201  ulimit -a
  202  ulimit -f
  203  ulimit -fS
  204  ulimit -f -S
  205  ulimit -f -H
  206  rm file.app
  207  ./write_limit file.app
  208  ll
  209  ulimit -f 1 -S
  210  ulimit -a
  211  ulimit -f -S
  212  ulimit -f -H
  213  rm file.app
  214  ./write_limit file.app
  215  ll
  216  rm file.app
  217  strace ./write_limit file.app
  218  ll
  219  vi write_seek.c
  220  ulimit -f unlimited -H
  221  sudo ulimit -f unlimited -H
  222  su
  223  ulimit -f 1024
  224  ulimit -f 
  225  ulimit -f -H
  226  ulimit -f unlimited
  227  su
  228  ulimit -S -a
  229  ulimit -H -a
  230  vi write_limit.c
  231  make
  232  vi write_limit.c
  233  make
  234  rm file.app
  235  ulimit -f 1 -S
  236  ulimit -f
  237  ./write_limit file.app
  238  rm file.app
  239  ls
  240  vi write_limit.c
  241  ll
  242  ulimit -S -f 1024
  243  vi write_limit.c
  244  make
  245  ls
  246  rm file.app
  247  ulimit -S -f 1
  248  ./write_limit file.app
  249  rm file.app
  250  strace ./write_limit file.app
  251  vi write_limit
  252  ulimit -S -f 1024
  253  vi write_limit.c
  254  make
  255  ulimit -S -f 1
  256  rm file.app 
  257  ./write_limit file.app
  258  ulimit -S -f 1024
  259  ulimit -S -f 1
  260  rm file.app
  261  ./write_limit file.app
  262  rm file.app
  263  strace ./write_limit file.app
  264  ll
  265  rm file.app
  266  ./write_limit file.app
  267  strace ./write_limit file.app
  268  ll
  269  vi copy.c
  270  ulimit -f unlimited
  271  su
  272  ulimit -a
  273  ulimit -f -H
  274  ulimit -f -H unlimited
  275  ulimit -H -f unlimited
  276  ulimit -S -f unlimited
  277  ulimit -a
  278  ll
  279  vi copy.c
  280  vi Makefile 
  281  make
  282  ./copy
  283  ls
  284  ls -lhrt
  285  copy < /dev/random > /dev/null
  286  ./copy < /dev/random > /dev/null
  287  ./copy < /dev/random
  288  ll
  289  dd if=/dev/random of=file.tst bs=807159 count=128
  290  ls -lhrt
  291  cat file.tst
  292  rm file.tst
  293  dd if=/dev/zero of=file.tst bs=807159 count=128
  294  exit
  295  g apue
  296  cd 02.chapter/
  297  ls
  298  ls -lhrt
  299  ./stat_fd /dev/random
  300  ./stat_fd /dev/zero 
  301  cd ..
  302  ll .
  303  ll -d .
  304  chmod u-x .
  305  ll -d .
  306  ls -a
  307  cd ..
  308  ll
  309  cd apue
  310  su
  311  ll .
  312  ll -l
  313  ls
  314  cd ..
  315  ll

04.chapter/ls.c
total 580K
-rw-rw-r--. 1 yunhai yunhai 8.4K 2016-11-09 04:43 ls.strace
-rw-rw-r--. 1 yunhai yunhai  60K 2016-11-09 04:43 ll.strace
-rw-rw-r--. 1 yunhai yunhai  905 2016-11-09 04:43 stat_fd.c
-rw-rw-r--. 1 yunhai yunhai  590 2016-11-09 04:44 umask_fd.c
-rw-rw-r--. 1 yunhai yunhai  942 2016-11-09 04:46 acc.c
-rw-r--r--. 1 yunhai yunhai  366 2016-11-09 04:46 chmod_fd.c
-rw-rw-r--. 1 yunhai yunhai  271 2016-11-09 04:47 hold.c
lrwxrwxrwx. 1 yunhai yunhai   12 2016-12-26 05:47 fake_entry -> nonexistence
lrwxrwxrwx. 1 yunhai yunhai    3 2016-12-26 05:47 entry -> tmp
drwxrwxr-x. 2 yunhai yunhai 4.0K 2017-01-03 05:03 tmp
-rw-rw-r--. 1 yunhai yunhai 4.4K 2017-01-03 05:16 stat_fd.o
-rw-rw-r--. 1 yunhai yunhai  17K 2017-01-03 05:16 apue.o
-rwxrwxr-x. 1 yunhai yunhai  20K 2017-01-03 05:16 stat_fd
-rw-rw-r--. 1 yunhai yunhai 3.6K 2017-01-03 05:16 acc.o
-rwxrwxr-x. 1 yunhai yunhai  19K 2017-01-03 05:16 acc
-rw-rw-r--. 1 yunhai yunhai 3.2K 2017-01-03 05:16 umask_fd.o
-rwxrwxr-x. 1 yunhai yunhai  19K 2017-01-03 05:16 umask_fd
-rw-rw-r--. 1 yunhai yunhai 4.0K 2017-01-03 05:16 chmod_fd.o
-rwxrwxr-x. 1 yunhai yunhai  20K 2017-01-03 05:16 chmod_fd
-rw-rw-r--. 1 yunhai yunhai 2.8K 2017-01-03 05:16 hold.o
-rwxrwxr-x. 1 yunhai yunhai  18K 2017-01-03 05:16 hold
-rwxr-xr-x. 1 yunhai yunhai 117K 2017-01-03 05:20 ls
-rw-r--r--. 1 yunhai yunhai 4.1K 2017-01-05 04:45 follow_links.c~
-rw-r--r--. 1 yunhai yunhai 4.3K 2017-01-05 04:45 follow_links.c
-rw-rw-r--. 1 yunhai yunhai 9.4K 2017-01-05 04:45 follow_links.o
-rwxrwxr-x. 1 yunhai yunhai  24K 2017-01-05 04:45 follow_links
lrwxrwxrwx. 1 yunhai yunhai   12 2017-01-05 04:45 fake_proxy -> nonexistence
-rwxrwxrwx. 1 yunhai yunhai 117K 2017-01-05 04:46 target
lrwxrwxrwx. 1 yunhai yunhai    6 2017-01-05 04:46 proxy -> target
-rw-r--r--. 1 yunhai yunhai  918 2017-01-16 05:13 Makefile~
-rw-r--r--. 1 yunhai yunhai 1.1K 2017-01-16 05:13 Makefile
-rw-rw-r--. 1 yunhai yunhai  365 2017-01-16 05:16 ftw.c~
-rw-rw-r--. 1 yunhai yunhai  743 2017-01-16 05:16 ftw.c
-rw-rw-r--. 1 yunhai yunhai 4.6K 2017-01-16 05:16 ftw.o
-rwxrwxr-x. 1 yunhai yunhai  20K 2017-01-16 05:16 ftw
-rwxrw-r--. 1 yunhai yunhai   81 2017-01-16 05:40 foo.sh
-rw-rw-r--. 1 yunhai yunhai    0 2017-01-19 04:46 ls.c
